src/main.rs

```diff
@@ -438,19 +240,16 @@ pub struct Mode {
     value: bool,
 }
 
-fn on_command(command_string: &str,
-              commands: &BTreeMap<String, Command>,
-              variables: &mut BTreeMap<String, String>,
-              modes: &mut Vec<Mode>) {
+fn on_command(command_string: &str, commands: &BTreeMap<String, Command>, shell: &mut Shell) {
     // Show variables
     if command_string == "$" {
-        for (key, value) in variables.iter() {
+        for (key, value) in shell.variables.iter() {
```

Comment:
It looks like the BTreeMap happens to keep it in alphabetical order. I will make sure that it sorts them when it prints them out when I change it to a hashmap.